test/CadlRanchProjects.Tests/type-property-nullable.cs

```diff
@@ -285,5 +289,175 @@ public Task Type_Property_Nullable_CollectionsString_patchNull() => Test(async (
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Array, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void NotRequiredNullablePropertiesSerializeWhenSet()
+        {
+            var inputModel = new CollectionsStringProperty();
+            inputModel.RequiredProperty = "required";
+            inputModel.NullableProperty = new List<string> { "1", "2" };
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual("2", element.GetProperty("nullableProperty")[1].GetString());
+        }
+
+        [Test]
+        public void NotRequiredNullablePropertiesDeserializedWithNulls()
+        {
+            var model = CollectionsStringProperty.DeserializeCollectionsStringProperty(JsonDocument.Parse("{\"NonRequiredNullableString\": null}").RootElement);
+            Assert.Null(model.NullableProperty);
+        }
+
+        [Test]
+        public void NullablePropertiesSerializedAsNulls()
+        {
+            var inputModel = new CollectionsStringProperty("required", null);
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Null, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void NullablePropertiesSerializedAsEmptyLists()
+        {
+            var inputModel = new CollectionsStringProperty("required", Array.Empty<string>());
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+            Assert.AreEqual(JsonValueKind.Array, element.GetProperty("nullableProperty").ValueKind);
+            Assert.AreEqual(0, element.GetProperty("nullableProperty").GetArrayLength());
+        }
+
+        [Test]
+        public void NullablePropertiesDeserializedAsNullsWithUndefined()
+        {
+            var model = CollectionsStringProperty.DeserializeCollectionsStringProperty(JsonDocument.Parse("{}").RootElement);
+            Assert.IsNull(model.NullableProperty);
+        }
+
+        [Test]
+        public void NullablePropertiesDeserializedAsUndefinedWithNulls()
+        {
+            var model = CollectionsStringProperty.DeserializeCollectionsStringProperty(JsonDocument.Parse("{\"nullableProperty\": null}").RootElement);
+            Assert.IsNotNull(model.NullableProperty);
+            Assert.IsFalse(!(model.NullableProperty is ChangeTrackingList<string> changeTrackingList1 && changeTrackingList1.IsUndefined));
+        }
+
+        [Test]
+        public void NullablePropertiesDeserializedAsValues()
+        {
+            var model = CollectionsStringProperty.DeserializeCollectionsStringProperty(JsonDocument.Parse("{\"nullableProperty\": [\"a\", \"b\"]}").RootElement);
+            Assert.AreEqual(new[] { "a", "b" }, model.NullableProperty);
+        }
+
+        [Test]
+        public void RequiredNullableCollectionsSerializeAsNull()
```

Comment:
we are using the word list in other test cases, we should not change.
```suggestion
        public void RequiredNullableListsSerializeAsNull()
```

Reply:
fixed